commands/uploader.go

```diff
@@ -24,10 +24,17 @@ type uploadContext struct {
 	committerName  string
 	committerEmail string
 
-	locks          map[string]locking.Lock
 	trackedLocksMu *sync.Mutex
-	ownedLocks     []locking.Lock
-	unownedLocks   []locking.Lock
+
+	// ALL verifiable locks
+	ourLocks   map[string]locking.Lock
+	theirLocks map[string]locking.Lock
+
+	// locks from ourLocks that were modified in this push
+	ownedLocks []locking.Lock
```

Comment:
Since this (and `unownedLocks`) are always subsets of `ourLocks` and `theirLocks` respectively, do you think that we could wrap the `locking.Lock` type here and store less copies of it in memory?

I was thinking:

```go
type owner byte

const (
        ownerAbsent owner = iota
        ownerOwned
        ownerUnowned
)

type lockWithOwner struct {
        locking.Lock

        State owner
}

ourLocks map[string]lockWithOwner
```